search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/OpenSearchChatConnector.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.llm;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.text.StringEscapeUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.Client;
+import org.opensearch.common.action.ActionFuture;
+import org.opensearch.ml.client.MachineLearningClient;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.MLInputDataset;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.output.MLOutput;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Wrapper for talking to LLMs via OpenSearch HttpConnector.
+ */
+public class OpenSearchChatConnector implements Llm {
+
+    private static final Logger log = LogManager.getLogger();
+
+    private static final String roleUser = "user";
+
+    private final String openSearchModelId;
+
+    private MachineLearningClient mlClient;
+
+    public OpenSearchChatConnector(String openSearchModelId, Client client) {
+        checkNotNull(openSearchModelId);
+        this.openSearchModelId = openSearchModelId;
+        this.mlClient = new MachineLearningNodeClient(client);
+
+    }
+
+    @VisibleForTesting
+    void setMlClient(MachineLearningClient mlClient) {
+        this.mlClient = mlClient;
+    }
+
+    /**
+     * Use ChatCompletion API to generate an answer.
+     *
+     * @param input
+     * @return
+     */
+    @Override
+    public ChatCompletionOutput createChatCompletion(ChatCompletionInput input) {
+
+        OpenSearchChatCompletionInput openAiInput = (OpenSearchChatCompletionInput) input;
```

Comment:
why `openAiInput`? Is this chat completion only tied to openAI?

Reply:
It's not tied to OpenAI.  I will rename it.